### DISCUSSION TITLE
How can I save Web3AuthCore in localstorage?

### DISCUSSION THREAD

**Discussion by RoniFinTech** [Web3Auth/Web3Auth#902] [Help]
So my goal is to maintain the state even after refreshing the browser and the only way I can think of is to save Web3AuthCore in local storage.

Any thoughts?

**Comment by shahbaz17** [comment]
Hey @RoniFinTech 

After successfully logging in, Does the login state lose after refreshing the browser?

Can you try [this demo](https://main.dkf4v281e9gg7.amplifyapp.com/)? And try to reproduce it? The code for this demo can be found on [this GitHub repository](https://github.com/Web3Auth/examples/tree/main/web-core-sdk/custom-authentication/auth0-react-core-example).

**Reply by RoniFinTech** [reply]
Hi @shahbaz17
Thanks for the answer.
Looks like you are right. the state is maintained after I log in to my app.
My issue was that my `login` is in a different component from `App`... so I need to pass Web3Auth object to the children...
I think I got it from here...

Thanks!
